File: iOSObjectiveCFoundation/Classes/Frontend/Component/Vfx/TimeLine.m
// ======================================================================
// Project Name    : ios_foundation
//
// Copyright © 2018 U-CREATES. All rights reserved.
//
// This source code is the property of U-CREATES.
// If such findings are accepted at any time.
// We hope the tips and helpful in developing.
// ======================================================================
#import "TimeLine.h"
@implementation TimeLine
@synthesize currentFrame;
@synthesize currentTime;
@synthesize rate;
- (id)init {
    self = [super init];
    self.currentFrame = 0.0f;
    self.currentTime = 0.0f;
    self.rate = 1.0f;
    return self;
}
- (void)next:(NSTimeInterval)delta {
    [self next:delta multipleRate:1.0f];
    return;
}
- (void)next:(NSTimeInterval)delta multipleRate:(float)multipleRate {
    self.currentFrame += self.rate;
    self.currentTime += (delta * multipleRate);
    return;
}
- (void)restore {
    self.currentFrame = 0.0f;
    self.currentTime = 0.0f;
    return;
}
@end

File: iOSSwiftFoundation/Classes/Frontend/Component/Vfx/TimeLine.swift
// ======================================================================
// Project Name    : ios_foundation
//
// Copyright © 2018 U-CREATES. All rights reserved.
//
// This source code is the property of U-CREATES.
// If such findings are accepted at any time.
// We hope the tips and helpful in developing.
// ======================================================================
import Foundation
public class TimeLine {
    public var currentFrame: Float!
    public var currentTime: Float!
    public var rate: Float!
    public init() {
        self.currentTime = 0.0
        self.currentFrame = 0.0
        self.rate = 1.0
    }
    open func next(delta: TimeInterval) -> Void {
        self.next(delta: delta, multipleRate: 1.0)
        return
    }
    open func next(delta: TimeInterval, multipleRate: Float) -> Void {
        self.currentFrame += self.rate
        self.currentTime += (Float(delta) * multipleRate)
        return
    }
    open func restore() -> Void {
        self.currentFrame = 0.0
        self.currentTime = 0.0
        return
    }
}
